pkg/crc/preflight/preflight_checks_windows.go

```diff
@@ -100,15 +100,43 @@ func checkHyperVServiceRunning() error {
 }
 
 func checkUserPartOfCrcUsersAndHypervAdminsGroup() error {
-	_, _, err := powershell.Execute(fmt.Sprintf("Get-LocalGroupMember -Group 'crc-users' -Member '%s'", username()))
+	groupMembers, _, err := powershell.Execute(`(Get-LocalGroupMember -Group 'crc-users').Name`)
 	if err != nil {
 		return err
 	}
-
+	logging.Debug("Checking current user part of 'crc-user' group")
+	if !usernameInMembersList(username(), groupMembers) {
+		return fmt.Errorf("Could not find: %s in 'crc-users' group", username())
+	}
 	// https://support.microsoft.com/en-us/help/243330/well-known-security-identifiers-in-windows-operating-systems
 	// BUILTIN\Hyper-V Administrators => S-1-5-32-578
-	_, _, err = powershell.Execute(fmt.Sprintf("Get-LocalGroupMember -SID 'S-1-5-32-578' -Member '%s'", username()))
-	return err
+	groupMembers, _, err = powershell.Execute(`(Get-LocalGroupMember -SID 'S-1-5-32-578').Name`)
+	if err != nil {
+		return err
+	}
+	logging.Debug("Checking current user part of 'Hyper-v Administrators' group")
+	if !usernameInMembersList(username(), groupMembers) {
+		return fmt.Errorf("Could not find: %s in 'Hyper-v Administrators' group", username())
+	}
+	return nil
+}
+
+func usernameInMembersList(username, members string) bool {
+	m := strings.Split(members, "\n")
+	var memberList []string
+	// remove any empty elements
+	for _, elem := range m {
+		if strings.TrimSpace(elem) != "" {
+			memberList = append(memberList, strings.TrimSpace(elem))
+		}
+	}
+	logging.Debugf("group members: %s", strings.Join(memberList, ","))
+	for _, member := range memberList {
+		if strings.Contains(member, username) {
```

Comment:
Couldn't this give false positives?
https://stackoverflow.com/questions/4548476/powershell-list-local-users-and-their-groups has this example:
```
PS C:\WINDOWS\system32> Get-LocalGroupMember -name users

ObjectClass Name                             PrincipalSource 
----------- ----                             --------------- 
User        DESKTOP-R05QDNL\someUser1        Local           
User        DESKTOP-R05QDNL\someUser2        MicrosoftAccount
Group       NT AUTHORITY\INTERACTIVE         Unknown  
```

I'm guessing the list we'll get in CRC is `DESKTOP-R05QDNL\someUser1`, `DESKTOP-R05QDNL\someUser2`?
If `username()` is `DESK` or `some`, `strings.Contains()` will succeed for both even if none of these users are in the list. I think you should be splitting the string on `\` and check the last part for equality?
Is there any go module/any Windows API we could use to get this information in an easier to consume format rather than spawning powershell commands to do it?

Reply:
> I'm guessing the list we'll get in CRC is DESKTOP-R05QDNL\someUser1, DESKTOP-R05QDNL\someUser2?
If username() is DESK or some, strings.Contains() will succeed for both even if none of these users are in the list. I think you should be splitting the string on \ and check the last part for equality?

yes yes, i missed this.

> Is there any go module/any Windows API we could use to get this information in an easier to consume format rather than spawning powershell commands to do it?

the golang `os/user` or `os` packages don't have anything to get the members of a group, i think we can get this information with the wmi library we already use (but that seems more complicated.. https://stackoverflow.com/a/26384427), similarly using win32 api from go would be complicated, powershell seems like the simplest